staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go

```diff
@@ -233,17 +245,43 @@ func (s *specProxier) getOpenAPIV3Root() handler3.OpenAPIV3Discovery {
 	s.rwMutex.RLock()
 	defer s.rwMutex.RUnlock()
 
-	merged := handler3.OpenAPIV3Discovery{
-		Paths: make(map[string]handler3.OpenAPIV3DiscoveryGroupVersion),
-	}
+	paths := make(map[string][]string)
 
+	// first collect paths to all apiServiceInfo, we expect duplicates for eg:
+	// authorization.openshift.io, security.openshift.io and quota.openshift.io
+	// which are backed by both CRD apiextension apiserver and openshift apiserver
 	for _, apiServiceInfo := range s.apiServiceInfo {
 		if apiServiceInfo.discovery == nil {
 			continue
 		}
 
 		for key, item := range apiServiceInfo.discovery.Paths {
-			merged.Paths[key] = item
+			paths[key] = append(paths[key], item.ServerRelativeURL)
+		}
+	}
+
+	merged := handler3.OpenAPIV3Discovery{
+		Paths: make(map[string]handler3.OpenAPIV3DiscoveryGroupVersion),
+	}
+
+	for key, delegationURLs := range paths {
+		// some apiservices can have duplicate paths in openapi v3 discovery (same path and hash) as they are backed by the same apiserver
+		delegationUniqueURLs := sets.List(sets.New(delegationURLs...))
+		// we either have just one url or a special URL like a /version
+		if len(delegationUniqueURLs) == 1 || (len(delegationUniqueURLs) > 1 && !strings.HasPrefix(delegationUniqueURLs[0], "/openapi/v3/apis/")) {
```

Comment:
As far as I understand we are trying to filter the "standard" ones. Can't we make this check `!strings.HasPrefix(delegationUniqueURLs[0], "/openapi/v3/apis/")` generic?.

Reply:
it shouldn't happen that we get mix of different prefixes, but ok, made the function generic

Reply:
there is also an issue with the original implementation when paths overlap, but have not made an issue/fix for this yet